src/core/core.controller.js

```diff
@@ -314,6 +328,7 @@ module.exports = function(Chart) {
 
 				if (meta.controller) {
 					meta.controller.updateIndex(datasetIndex);
+					meta.controller.linkScales();
```

Comment:
I came across this when I tried to update scales with different `id` - `dataset._meta` contains scales id which didn't update.
I check the scales id of `meta` here, relink the scale if discarded.

I'm not sure whether I missed anything else, please point out if there are. It's a little bit too many configs here : P